components/checkbox/index.css

```diff
@@ -24,9 +157,299 @@ governing permissions and limitations under the License.
+   * while using the colors of a default checkbox
+   */
+  &.is-readOnly {
+    border-color: var(--highcontrast-checkbox-color-default, var(--mod-checkbox-control-selected-color-default, var(--spectrum-checkbox-control-selected-color-default)));
+
+    &:hover {
+      .spectrum-Checkbox-box {
+        &:before {
+          border-color: var(--highcontrast-checkbox-color-default, var(--mod-checkbox-control-selected-color-default, var(--spectrum-checkbox-control-selected-color-default)));
+        }
+      }
+    }
+
+    &:active {
+      .spectrum-Checkbox-box {
+        &:before {
+          border-color: var(--highcontrast-checkbox-selected-color-default, var(--mod-checkbox-control-selected-color-default, var(--spectrum-checkbox-control-selected-color-default)));
+        }
+      }
+    }
+  }
+
+  &.is-readOnly .spectrum-Checkbox-input,
+  &.is-readOnly .spectrum-Checkbox-input:checked {
+    &:disabled + .spectrum-Checkbox-box {
+      &:before {
+        border-color: var(--highcontrast-checkbox-color-default, var(--mod-checkbox-control-selected-color-default, var(--spectrum-checkbox-control-selected-color-default)));
+        background-color: var(--highcontrast-checkbox-background-color-default, var(--mod-checkbox-checkmark-color, var(--spectrum-checkbox-checkmark-color)));
+      }
+    }
+
+    &:disabled ~ .spectrum-Checkbox-label {
+      forced-color-adjust: none;
+      color: var(--highcontrast-checkbox-color-default, var(--mod-checkbox-content-color-default, var(--spectrum-checkbox-content-color-default)));
+    }
+  }
+
+  /* Indeterminate */
+  &.is-indeterminate {
+    .spectrum-Checkbox-box,
+    .spectrum-Checkbox-input:checked + .spectrum-Checkbox-box {
+      &:before {
+        border-color: var(--highcontrast-checkbox-highlight-color-default, var(--mod-checkbox-control-selected-color-default, var(--spectrum-checkbox-control-selected-color-default)));
+        border-width: calc(var(--spectrum-checkbox-control-size) / 2);
+      }
+
+      .spectrum-Checkbox-checkmark {
+        display: none;
+      }
+
+      .spectrum-Checkbox-partialCheckmark {
+        display: block;
+
+        transform: scale(1);
+        opacity: 1;
+      }
+    }
+
+    &:hover {
+      .spectrum-Checkbox-input:checked + .spectrum-Checkbox-box,
+      .spectrum-Checkbox-box {
+        &:before {
+          border-color: var(--highcontrast-checkbox-highlight-color-hover, var(--mod-checkbox-control-selected-color-hover, var(--spectrum-checkbox-control-selected-color-hover)));
+        }
+      }
+    }
+  }
+
+  /* Invalid Indetermiante */
+  &.is-invalid.is-indeterminate {
+    .spectrum-Checkbox-box,
+    .spectrum-Checkbox-input:checked + .spectrum-Checkbox-box {
+      &:before {
+        border-color: var(--highcontrast-checkbox-color-default, var(--mod-checkbox-invalid-color-default, var(--spectrum-checkbox-invalid-color-default)));
+        border-width: calc(var(--spectrum-checkbox-control-size) / 2);
```

Comment:
I noted it above, but let's use the abstracted custom property that you create and also add the `--mod-` custom property here.